### PR TITLE
Add AdgenerationBidAdapter

### PR DESCRIPTION
**Pull request by adg-dev**
### Type of change
- [x] New bidder adapter

### Description of change
Adding Adgeneraton bid adapter for Prebid 1.0.
- test parameters for validation bids.
```Javascript
{
  bidder: 'adg',
  params: {
    id: '58278',
    width: '300',
    height: '250'
  }
}

```
- Maintainer: ssp-ope@supership.jp
- [x] official adapter submission

### CODE REVIEW

File: modules/adgenerationBidAdapter.js [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+   * Make a server request from the list of BidRequests.
+   *
+   * @param {validBidRequests[]} - an array of bids
+   * @return ServerRequest Info describing the request to the server.
+   */
+  buildRequests: function (validBidRequests) {
+    let serverRequests = [];
+    for (let i = 0, len = validBidRequests.length; i < len; i++) {
+      const validReq = validBidRequests[i];
+      const url = validReq.params.debug ? 'http://api-test.scaleout.jp/adsv/v1' : 'https://d.socdm.com/adsv/v1';
```
Comment (mkendall07) [inline modules/adgenerationBidAdapter.js]: prefer these strings to be constants.

File: modules/adgenerationBidAdapter.js [hunk 2]
```diff
@@ -0,0 +1,194 @@
+import * as utils from 'src/utils';
+// import {config} from 'src/config';
+import {registerBidder} from 'src/adapters/bidderFactory';
+import { NATIVE } from 'src/mediaTypes';
+const ADG_BIDDER_CODE = 'adgeneration';
+
+export const spec = {
+  code: ADG_BIDDER_CODE,
+  aliases: ['adg'], // short code
+  supportedMediaTypes: [NATIVE],
```
Comment (mkendall07) [inline modules/adgenerationBidAdapter.js]: If you are supporting more than native, I.E. `banner`, you need to set it explicitly here.

File: modules/adgenerationBidAdapter.js [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+}
+
+function insertVASTMethod(targetId, vastXml) {
+  let apvVideoAdParam = {
+    s: targetId
+  };
+  let script = document.createElement(`script`);
+  script.type = 'text/javascript';
+  script.innerHTML = `(function(){ new APV.VideoAd(${JSON.stringify(apvVideoAdParam)}).load('${vastXml.replace(/\r?\n/g, '')}'); })();`;
+  return script.outerHTML;
```
Comment (mkendall07) [inline modules/adgenerationBidAdapter.js]: can you explain what this is doing/why it's needed?

**Comment by mkendall07** [comment]
@adg-dev 
Thanks for the PR. Left a few comments. Also please submit a PR to the [docs repo](https://github.com/prebid/prebid.github.io) to add a file for your adapter to the [bidders directory](https://github.com/prebid/prebid.github.io/tree/master/dev-docs/bidders) so your adapter's params will appear on the [bidders page](http://prebid.org/dev-docs/bidders.html).

File: modules/adgenerationBidAdapter.js (hunk 3 above)
Comment (banakemi) [reply modules/adgenerationBidAdapter.js]: VAST data, in our sever responses , is converted by CSS Sprites etc.

**Comment by banakemi** [comment]
Hi, @mkendall07 
Thanks for review.
Fix your comments and our docs. (https://github.com/prebid/prebid.github.io/pull/575)

**Comment by mkendall07** [comment]
@banakemi 
Thanks. Can you please drop `width` and `height` from your params? These can be obtained from the bid request directly.

**Comment by banakemi** [comment]
@mkendall07 
Thanks!
...Are you talking about interpret Response
methods?
If so, I guess those params are needed to display the banner. 
Could you tell me which part you're talking about, please?

**Comment by mkendall07** [comment]
I mean from your bidder params:
```
params: {
    id: '58278',
    width: '300',
    height: '250'
  }
```

The adunit will have these defined already

**Pull request merged**